src/util/kaldi-table-inl.h

```diff
@@ -178,20 +191,35 @@ template<class Holder>  class SequentialTableReaderScriptImpl:
       else  // orig_state_ was kLoadFailed, which only could have happened
         // if the user called FreeCurrent().
         KALDI_ERR << "You called Value() after FreeCurrent().";
-    } else if (state_ != kLoadSucceeded) {
+    } else if (state_ == kRangeExtractionFailed) {
+      KALDI_ERR << "Failed to load object from "
+                << PrintableRxfilename(data_rxfilename_) 
+                << " [" << range_ << "] "
+                << " (to suppress this error, add the permissive "
+                << "(p, ) option to the rspecifier.";
+    } else if (state_ != kLoadSucceeded && state_ != kRangeExtracted) {
       // This would be a coding error.
       KALDI_ERR << "Value() called at the wrong time.";
     }
-    return holder_.Value();
+    if (state_ == kRangeExtracted) {
+      return range_holder_.Value();
+    } else {
+      // Here state_ is kLoadSucceeded.
+      return holder_.Value();
+    }
   }
+
   void FreeCurrent() {
-    if (state_ == kLoadSucceeded) {
+    if (state_ == kLoadSucceeded || 
```

Comment:
Here, FreeCurrent() should only be called when we successfully have a value, so state kRangeExtractionFailed isn't expected in non-error operation.  Actually, if state_ == kLoadSucceeded [i.e. there is no range object], we should clear holder_, otherwise if state is kRangeExtracted then clear range_holder_, otherwise print a warning.